test/e2e_flags.go

```diff
@@ -73,9 +90,45 @@ func initializeFlags() *EnvironmentFlags {
 
 	flag.StringVar(&f.Tag, "tag", "latest", "Provide the version tag for the test images.")
 
+	klog.InitFlags(nil)
+	flag.Set("v", klogDefaultLogLevel)
+	flag.Set("alsologtostderr", "true")
+
 	return &f
 }
 
+func printFlags() {
+	fmt.Print("Test Flags: {")
+	flag.CommandLine.VisitAll(func(f *flag.Flag) {
+		fmt.Printf("'%s': '%s', ", f.Name, f.Value.String())
+	})
+	fmt.Println("}")
+}
+
+// SetupLoggingFlags initializes the logging libraries at runtime
+func SetupLoggingFlags() {
+	flagsMutex.Lock()
+	defer flagsMutex.Unlock()
```

Comment:
As I mentioned in the other PR ( I am still not sure how they relate to each other), should we use `sync.Once` here, rather than lock/unlock? This is supposed to be called once anyway, right?

Reply:
I can't test that it doesn't get called during package initialization then without adding much more variables that just end up being the same thing + extra boilerplate

Reply:
Sorry, I don't really parse your comment.
sync.Once will verify it is called once. _do we need_ it to be called more than once?

Reply:
Yes, we do for the unit test, but I could just override the sync.Once variable to make that happen.

But I want to verify this function does NOT get called during package initialization, and sync.Once does not provide any way to see if it has been triggered. And if you add in another variable to detect if it's been triggered, it ends up looking nearly the same as this but worse.

Reply:
(because flag parsing happening at package initialization time is why it broke with Go 1.13 and the fix didn't actually fix it)

Reply:
HOw would you do it, if your `sync.Once` variable is package private?

Reply:
The unit test is in the same package

Reply:
Ah, yeah, but I care more about the code that goes to the library, than the bloat in the unit test.

Reply:
The unit test won't get any worse, the library will be much worse, because it will still need to set a value saying it ran, like it does now, then define the sync struct and wrap all work into an anonymous function into syncVariable.Do(func () {...
So it ends up being more lines and indentation to achieve the same thing.

Reply:
OK.

Reply:
If sync.Once said whether it had been run before, it would be cleaner, but it doesn't ....
Oh, wait, it doesn't matter what function you call, so I could call it with anything...
hold on...

Reply:
okay, done.